src/scene/container/RenderGroupPipe.ts

```
@@ -26,25 +28,54 @@ export class RenderGroupPipe implements InstructionPipe<RenderGroup>
 
     public addRenderGroup(renderGroup: RenderGroup, instructionSet: InstructionSet): void
     {
-        this._renderer.renderPipes.batch.break(instructionSet);
+        // TODO should we skip this if renderGroup is - render just in time?
+        // here! we need to render to a texture if required...
+        if (renderGroup.cacheAsTexture)
+        {
+            const batchableRenderGroup = renderGroup._batchableRenderGroup ??= BigPool.get(BatchableSprite);
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)